src/clj/rems/application/cleaner.clj

```diff
@@ -0,0 +1,46 @@
+(ns rems.application.cleaner
+  (:require [mount.core :as mount]
+            [clj-time.core :as time]
+            [clojure.test :refer :all]
+            [clojure.tools.logging :as log]
+            [rems.scheduler :as scheduler]
+            [rems.db.applications :as applications]))
+
+(defn- is-expired-draft?
+  [application]
+  (let [last-activity (:application/last-activity application)
+        state (:application/state application)]
+    (and (= :application.state/draft state)
+         (time/before? last-activity (time/minus (time/now) (time/days 90))))))
+
+(deftest test-is-expired-draft
+  (testing "should identify expired draft application"
+    (are [expected input] (= expected (is-expired-draft? input))
+      true {:application/state :application.state/draft
+            :application/last-activity (time/minus (time/now) (time/days 90) (time/seconds 1))}
+      false {:application/state :application.state/draft
+             :application/last-activity (time/now)}
+      false {:application/state :application.state/submitted
+             :application/last-activity (time/minus (time/now) (time/days 90) (time/seconds 1))})))
+
+(defn process-expired-draft-applications! []
+  (let [removed-ids (atom [])
+        failed-ids (atom [])]
+    (doseq [app-id (->> (applications/get-all-unrestricted-applications)
+                        (filter is-expired-draft?)
+                        (map :application/id))]
+      (try
+        (applications/remove-application-data! app-id)
```

Comment:
command `:application.command/delete` would be nice due to the standard API, but under the hood it would utilize `applications/delete-application!` which performs the same deletion steps, but reloads cache, effectively after each delete. this was deemed, as discussed, not ideal, so I extracted application deletion steps into common functionality which is used here.